doc/source/ray-core/examples/map_reduce.ipynb

```diff
@@ -0,0 +1,430 @@
+    "   For example, four instances of `(word, 1)` combine for a final count of `word: 4`."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "87bd5e5e",
+   "metadata": {},
+   "source": [
+    "The first and last phases are in the MapReduce name, but the middle phase is equally crucial.\n",
+    "These phases appear straightforward, but their power is in running them concurrently on multiple machines.\n",
+    "This figure illustrates the three MapReduce phases on a set of documents:"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0d0b1af6",
+   "metadata": {},
+   "source": [
+    "\n",
+    "![Simple Map Reduce](https://raw.githubusercontent.com/maxpumperla/learning_ray/main/notebooks/images/chapter_02/map_reduce.png)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "2225ae60",
+   "metadata": {},
+   "source": [
+    "## Loading Data\n",
+    "\n",
+    "We use Python to implement the MapReduce algorithm for the word count and Ray to parallelize the computation.\n",
+    "We start by loading some sample data from the Zen of Python, a collection of coding guidelines for the Python community. Access to the Zen of Python, according to Easter egg tradition, is by typing `import this` in a Python session. \n",
+    "We divide the Zen of Python into three separate \"documents\" by treating each line as a separate entity\n",
+    "and then splitting the lines into three partitions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "91c6ddc0",
+   "metadata": {
+    "lines_to_next_cell": 2
+   },
+   "outputs": [],
+   "source": [
+    "import subprocess\n",
+    "zen_of_python = subprocess.check_output([\"python\", \"-c\", \"import this\"])\n",
+    "corpus = zen_of_python.split()\n",
+    "\n",
+    "num_partitions = 3\n",
+    "chunk = len(corpus) // num_partitions\n",
+    "partitions = [\n",
+    "    corpus[i * chunk: (i + 1) * chunk] for i in range(num_partitions)\n",
+    "]"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "1357e924",
+   "metadata": {},
+   "source": [
+    "## Mapping Data\n",
+    "\n",
+    "To determine the map phase, we require a map function to use on each document.\n",
+    "The output is the pair `(word, 1)` for every word found in a document.\n",
+    "For basic text documents we load as Python strings, the process is as follows:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "id": "742193e2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def map_function(document):\n",
+    "    for word in document.lower().split():\n",
+    "        yield word, 1"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "52009879",
+   "metadata": {},
+   "source": [
+    "We use the `apply_map` function on a large collection of documents by marking it as a task in Ray using the `@ray.remote` decorator.\n",
```

Comment:
can we link to ray.remote api ref. I think we want to link to api ref at least for the first occurrence of a ray function.

Reply:
done